repos/spack_repo/builtin/packages/py_geopmpy/package.py

```diff
@@ -13,30 +13,48 @@ class PyGeopmpy(PythonPackage):
 
     homepage = "https://geopm.github.io"
     git = "https://github.com/geopm/geopm.git"
-    url = "https://github.com/geopm/geopm/tarball/v3.1.0"
+    url = "https://github.com/geopm/geopm/tarball/v3.2.0"
 
     maintainers("bgeltz", "cmcantalupo")
     license("BSD-3-Clause")
     tags = ["e4s"]
 
     version("develop", branch="dev", get_full_repo=True)
+    version("3.2.0", sha256="b708233e1bfda66408c500f2ac0cbaf042140870bffdced12dd7cabbd18e0025")
     version("3.1.0", sha256="2d890cad906fd2008dc57f4e06537695d4a027e1dc1ed92feed4d81bb1a1449e")
 
+    for ver in ["3.1.0", "3.2.0", "develop"]:
+        depends_on(f"py-geopmdpy@{ver}", type="run", when=f"@{ver}")
+        depends_on(f"geopm-runtime@{ver}", type=("build", "run"), when=f"@{ver}")
+
     depends_on("python@3.6:3", type=("build", "run"))
-    depends_on("py-setuptools@53.0.0:", type="build")
+    depends_on("py-setuptools@53.0.0:", when="@3.1", type="build")
+    depends_on("py-setuptools@59.6.0:", when="@3.2:", type="build")
     depends_on("py-setuptools-scm@7.0.3:", when="@3.1:", type="build")
     depends_on("py-build@0.9.0:", when="@3.1:", type="build")
-    depends_on("py-cffi@1.14.5:", type="run")
+    depends_on("py-cffi@1.14.5:", when="@3.1.0", type="run")
+    depends_on("py-cffi@1.14.5:", when="@3.2:", type=("build", "run"))
```

Comment:
This is also newer than what is specified in `pyproject.toml`.

Reply:
The reason for this discrepancy has to do with building RPMs for geopmdpy and geopmpy:
The `pyproject.toml` is used in the rpmbuild process for some distros.  Because of this, and because on those systems the newest version of py-cffi that's available in the package repositories is 1.13.2 we must use that version.

When we install through Spack or pip, we'd prefer to use 1.14.5 or newer, which is why 1.14.5 is what's in the requirements.txt.

If you look at the `setup.cfg` for geopmdpy or geopmpy, you'll see the even older 1.11.5 version specified.  It's required for similar old distro support reasons.